tests/Microsoft.Bot.Streaming.Tests/Payloads/PayloadAssemblerTests.cs

```diff
@@ -71,5 +90,309 @@ public void PayloadAssembler_Close__DoesNotSetEnd()
 
             Assert.False(a.End);
         }
+
+        [Fact]
+        public void PayloadAssemblerManager_GetPayloadStream_Request_Returns_Stream()
+        {
+            var id = Guid.NewGuid();
+            var streamManager = new StreamManager();
+
+            var header = new Header { Id = id, Type = PayloadTypes.Request };
+
+            static Task OnReceiveRequest(Guid id, ReceiveRequest req) => Task.CompletedTask;
+            static Task OnReceiveResponse(Guid id, ReceiveResponse res) => Task.CompletedTask;
+
+            var assembler = new PayloadAssemblerManager(streamManager, OnReceiveRequest, OnReceiveResponse);
+            var stream = assembler.GetPayloadStream(header);
+
+            Assert.NotNull(stream);
+        }
+
+        [Fact]
+        public void PayloadAssemblerManager_GetPayloadStream_Response_Returns_Stream()
+        {
+            var id = Guid.NewGuid();
+            var streamManager = new StreamManager();
+
+            var header = new Header { Id = id, Type = PayloadTypes.Response };
+
+            static Task OnReceiveRequest(Guid id, ReceiveRequest req) => Task.CompletedTask;
+            static Task OnReceiveResponse(Guid id, ReceiveResponse res) => Task.CompletedTask;
+
+            var assembler = new PayloadAssemblerManager(streamManager, OnReceiveRequest, OnReceiveResponse);
+            var stream = assembler.GetPayloadStream(header);
+
+            Assert.NotNull(stream);
+        }
+
+        [Fact]
+        public void PayloadAssemblerManager_GetPayloadStream_With_No_Header_Type_Returns_Null()
+        {
+            var id = Guid.NewGuid();
+            var streamManager = new StreamManager();
+
+            var header = new Header { Id = id };
+
+            static Task OnReceiveRequest(Guid id, ReceiveRequest req) => Task.CompletedTask;
+            static Task OnReceiveResponse(Guid id, ReceiveResponse res) => Task.CompletedTask;
+
+            var assembler = new PayloadAssemblerManager(streamManager, OnReceiveRequest, OnReceiveResponse);
+            var stream = assembler.GetPayloadStream(header);
+
+            Assert.Null(stream);
+        }
+
+        [Fact]
+        public void PayloadAssemblerManager_GetPayloadStream_With_Active_Assembler_Returns_Null()
+        {
+            var id = Guid.NewGuid();
+            var streamManager = new StreamManager();
+
+            var header = new Header { Id = id, Type = PayloadTypes.Request };
+
+            static Task OnReceiveRequest(Guid id, ReceiveRequest req) => Task.CompletedTask;
+            static Task OnReceiveResponse(Guid id, ReceiveResponse res) => Task.CompletedTask;
+
+            var assembler = new PayloadAssemblerManager(streamManager, OnReceiveRequest, OnReceiveResponse);
+            var stream = assembler.GetPayloadStream(header);
+            stream = assembler.GetPayloadStream(header);
```

Comment:
Is this ok? The stream variable is assigned twice.

Reply:
There is a condition inside the GetPayloadStream, that when there is an active assembler, it returns null, this process is just to cover that return line.
![imagen](https://user-images.githubusercontent.com/62260472/127489318-7a320edd-53b8-4150-934f-d83d6fb07448.png)